se3cnn/SO3.py

```diff
@@ -478,32 +476,43 @@ def spherical_harmonics_coeff_to_sphere(coeff, alpha, beta):
     return s
 
 
-
 ################################################################################
 # Linear algebra
 ################################################################################
 
-def get_matrix_kernel(A, eps=1e-10):
-    """
-    Compute an orthonormal basis of the kernel (x_1, x_2, ...)
-    A x_i = 0
-    scalar_product(x_i, x_j) = delta_ij
-
-    :param A: matrix
-    :return: matrix where each row is a basis vector of the kernel of A
-    """
-    _u, s, v = torch.svd(A)
-
-    # A = u @ torch.diag(s) @ v.t()
-    kernel = v.t()[s < eps]
-    return kernel
-
-
-def get_matrices_kernel(As, eps=1e-10):
-    """
-    Computes the commun kernel of all the As matrices
-    """
-    return get_matrix_kernel(torch.cat(As, dim=0), eps)
+def get_d_null_space(l1, l2, l3, eps=1e-10):
+    import scipy
+    import scipy.linalg
+    import gc
+
+    def _DxDxD(a, b, c):
+        D1 = irr_repr(l1, a, b, c)
+        D2 = irr_repr(l2, a, b, c)
+        D3 = irr_repr(l3, a, b, c)
+        return torch.einsum('il,jm,kn->ijklmn', (D1, D2, D3)).view(n, n)
+
+    n = (2 * l1 + 1) * (2 * l2 + 1) * (2 * l3 + 1)
+    random_angles = [
+        [4.41301023, 5.56684102, 4.59384642],
+        [4.93325116, 6.12697327, 4.14574096],
+        [0.53878964, 4.09050444, 5.36539036],
+        [2.16017393, 3.48835314, 5.55174441],
+        [2.52385107, 0.29089583, 3.90040975],
+    ]
+
+    B = torch.zeros((n, n))                                                                             # preallocate memory
```

Comment:
no need to preallocate => `B = 0`

Reply:
while technically it is true that first `+=` is going to reassign value with such initialization, I am a little bit cautious about impact on ref count in `del`

Reply:
as you wish